nemo_aligner/algorithms/supervised.py

```diff
@@ -189,6 +197,9 @@ def fit(self):
                     metrics = {k: torch.as_tensor(v) for k, v in metrics.items()}
                     self.save(metrics, is_train_end=is_train_end)
 
+                if run_time_exceeded:
+                    return
```

Comment:
1. Can you add a log that says why we stop
2. Should we call `self.logger.finalize()` before returning?

Reply:
1. 07e3990e5ed9ec54d37d3c92e2afda9ee88fd3f4
2. I don't think we need to, the only time we finalize is when the run is successful(which I think here maybe we should consider it as not because we didn't finish training)

Reply:
Ok sounds good!